controllers/yawol-controller/loadbalancerset/helper.go

```diff
@@ -0,0 +1,93 @@
+package loadbalancerset
+
+import (
+	"fmt"
+
+	yawolv1beta1 "github.com/stackitcloud/yawol/api/v1beta1"
+	"github.com/stackitcloud/yawol/internal/helper"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+var relevantLBMConditionslForLBS = sets.New[helper.LoadbalancerCondition](
+	helper.ConfigReady,
+	helper.EnvoyReady,
+	helper.EnvoyUpToDate,
+)
+
+// areRelevantConditionsMet returns if all required conditions (from the
+// perspective of the LoadBalancerSet) are both `True` and up-to-date, according
+// to the passed expiration time.
+func areRelevantConditionsMet(machine *yawolv1beta1.LoadBalancerMachine, expiration metav1.Time, checkTransition bool) (bool, error) {
+	if machine.Status.Conditions == nil {
+		return false, fmt.Errorf("no conditions set")
+	}
+	found := 0
+	for _, condition := range *machine.Status.Conditions {
+		if !relevantLBMConditionslForLBS.Has(helper.LoadbalancerCondition(condition.Type)) {
+			continue
+		}
+		found++
+		transitionCheck := true
+		if checkTransition {
+			transitionCheck = condition.LastTransitionTime.Before(&expiration)
+		}
+		if transitionCheck && condition.Status != corev1.ConditionTrue {
+			return false, fmt.Errorf(
+				"condition: %v, reason: %v, status: %v, message: %v, lastTransitionTime: %v - %w",
+				condition.Type, condition.Reason, condition.Status, condition.Message, condition.LastTransitionTime,
+				helper.ErrConditionsNotInCorrectState,
+			)
+		}
+		if condition.LastHeartbeatTime.Before(&expiration) {
+			return false, fmt.Errorf(
+				"no condition heartbeat in the last 5 min: %w",
+				helper.ErrConditionsLastHeartbeatTimeToOld,
+			)
+		}
+	}
+
+	if found != len(relevantLBMConditionslForLBS) {
```

Comment:
Can we return an error indicating which condition is actually missing.

I would implement it the other way round:
Instead of looping over the present conditions, we should loop over the relevant conditions, look them up, and check if they have the required state.